### PR TITLE
Enable some new Dart VM features

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/7579

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth @Hixie

File: runtime/dart_init.cc [hunk 1]
```diff
@@ -100,6 +100,7 @@ static const char* kDartCheckedModeArgs[] = {
     "--enable_type_checks",
     "--error_on_bad_type",
     "--error_on_bad_override",
+    "--assert_initializer",
```
Comment (abarth) [inline runtime/dart_init.cc]: Does this actually enable to asserts or just configure the parser to accept the syntax?  I would have expected it to go in kDartLanguageArgs.

File: runtime/dart_init.cc [hunk 2]
```diff
@@ -78,18 +78,18 @@ const char kSnapshotAssetKey[] = "snapshot_blob.bin";
 
 namespace {
 
-static const char* kDartMirrorsArgs[] = {
+// Arguments passed to the Dart VM in all configurations.
+static const char* kDartLanguageArgs[] = {
     "--enable_mirrors=false",
+    "--background_compilation",
+    "--await_is_keyword",
+    "--warn_super",
```
Comment (abarth) [inline runtime/dart_init.cc]: What is this flag?

File: runtime/dart_init.cc (hunk 1 above)
Comment (jason-simmons) [inline runtime/dart_init.cc]: I believe it's the latter, and it makes sense to turn it on for all JIT mode configurations given that JIT doesn't always imply checked mode.  Moved it to kDartLanguageArgs.

File: runtime/dart_init.cc (hunk 2 above)
Comment (jason-simmons) [inline runtime/dart_init.cc]: Dart described it as "Warning if super initializer not last in initializer list."

Removing this for now

Comment (Hixie) [inline runtime/dart_init.cc]: It's a warning for when you have `super` not last in your initializer list. Pretty sure we always want it last; IIRC strong mode is going to require that anyway.

Comment (Hixie) [inline runtime/dart_init.cc]: warn_super should probably go in the checked mode list, as opposed to always on.

Comment (abarth) [inline runtime/dart_init.cc]: I'm not sure we should be enforcing warn_super at runtime.  We already have it at analysis time, which seems proper.  If we enforce it at runtime, we won't be able to run code from packages that fail this check.  If it's going to be part of strong mode, let's turn on runtime enforcement together with the rest of the dart ecosystem then.

**Comment by abarth** [comment]
I'm not sure what the consequences of await-is-keyword are, but we can try this.

**Comment by abarth** [comment]
LGTM

**Pull request merged**